tools/c7n_azure/c7n_azure/resources/access_control.py

```diff
@@ -32,12 +41,48 @@ class resource_type(object):
         get_spec = ('role_assignments', 'get_by_id', None)
         id = 'id'
         default_report_fields = (
+            'principalName',
+            'displayName',
+            'aadType',
             'name',
             'type',
             'properties.scope',
             'properties.roleDefinitionId'
         )
 
+    def augment(self, resources):
+        s = local_session(self.session_factory)
+        cred, sub_id = get_azure_cli_credentials(resource='https://graph.windows.net')
```

Comment:
I doubt this is OK  ;)

Reply:
Actually that didn't work and I found this post from Laurent himself on stackoverflow on how to get the right credentials: https://stackoverflow.com/questions/50223043/graphrbacmanagementclient-applications-create-returns-access-token-missing-or

Reply:
I know from previous experience with working with the graph that you need a special graph access token. There might another way but this was pretty simple and recommended by Laurent so I'd be surprised if we're not supposed to do this :P

Reply:
We make that same call you just need to pass the right resource, we'd just add it as a parameter on the session.  Your code clearly only runs if the user is running with CLI auth which isn't going to work.

Reply:
Man our session.py will need some work to get this going. What I have so far is still not working for even just the service principal credentials. I pass the resource essentially like this where resource is 'https://graph.windows.net':
```self.credentials = ServicePrincipalCredentials(
                    client_id=os.environ['AZURE_CLIENT_ID'],
                    secret=os.environ['AZURE_CLIENT_SECRET'],
                    tenant=os.environ['AZURE_TENANT_ID'],
                    resource=resource
                )
```
Even basically hard coded the client call to look like below since Graph client needs tenant id and not subscription id:   

```def client(self, client, resource=None):
        self._initialize_session(resource)
        service_name, client_name = client.rsplit('.', 1)
        svc_module = importlib.import_module(service_name)
        klass = getattr(svc_module, client_name)
        if resource == 'https://graph.windows.net':
            return klass(self.credentials, self.tenant_id)
        return klass(self.credentials, self.subscription_id)```

 And I still get errors when making the same call. Right now I'm getting a 403 Forbidden.